Reverse depth & 32F buffer format

AtomicWalrus opened the pull request:
-Adds reversed depth projection model, significantly increasing depth buffer effective resolution. Same model used in modern Unity & Unreal Engines.
-Adds 32F depth 8U stencil format GFXFormatD32FS8X24 (following DX naming conventions). Note this is a 64-bit format, and likely not suitable for mobile platforms. Revert to GFXFormatD24S8 in renderManager.tscript for mobile & "ancient" platforms. 
-Disables OpenGL-specific projection matrix code, as OGL now uses the same depth value range (1-0) as D3D.
-Corrects alignment of texture format details array.

AtomicWalrus committed:
Reverse depth & 32F buffer format

-Adds reversed depth projection model, dramatically increasing depth buffer effective resolution.
-Adds 32F depth 8U stencil format GFXFormatD32FS8X24 (following DX naming conventions). Note this is a 64-bit format, and likely not suitable for mobile platforms. Revert to GFXFormatD24S8 in renderManager.tscript for mobile & "ancient" platforms.
-Corrects alignment of texture type details array.

JeffProgrammer commented:
Mac does not support glClipControl. this will break mac support. glClipControl is from OpenGL 4.5

AtomicWalrus committed:
Fix for MacOS OGL 4.1 support

Workaround: MacOS does not have access to glClipControl.

Because the lower range of the 32F buffer is not available using this workaround, the improvements to depth behavior from depth reversal will be limited on MacOS. Behavior is equivalent to reversed depth on a 24UINT buffer; Better than stock, but not dramatically.

AtomicWalrus commented:
Added a conditional workaround for MacOS.

Unfortunately it's only partially equivalent, so while it's functional (near/far planes are correct) it won't produce the same improvement to depth performance. Looks a bit better than stock, but not by much.
